tests/unit/test_sns_listener.py

```diff
@@ -32,12 +35,80 @@ def test_unsubscribe_should_remove_listener(self):
             topic_arn,
             'http://localhost:1234/listen',
             'http',
-            sub_arn
+            sub_arn,
+            {}
         )
         self.assertTrue(sns_listener.get_subscription_by_arn(sub_arn))
         sns_listener.do_unsubscribe(sub_arn)
         self.assertFalse(sns_listener.get_subscription_by_arn(sub_arn))
 
+    def test_subscribe_should_parse_attributes(self):
+        topic_arn = 'arn:aws:sns:us-east-1:123456789012:test-topic'
+        sub_arn = 'arn:aws:sns:us-east-1:123456789012:test-topic:abc'
+
+        req_data = b'Action=Subscribe&' \
+                   b'Version=2010-03-31' \
+                   b'&TopicArn=arn%3Aaws%3Asns%3Aus-east-1%3A123456789012%3Atest-topic' \
+                   b'&Protocol=sqs' \
+                   b'&Endpoint=arn%3Aaws%3Asqs%3Aus-east-1%3A123456789012%3Atest-sqs' \
+                   b'&Attributes.entry.1.key=RawMessageDelivery' \
+                   b'&Attributes.entry.1.value=true'
+        headers = {'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8'}
+        response = Response()
+        response._content = \
+            b'<SubscribeResponse xmlns="http://sns.amazonaws.com/doc/2010-03-31/">\n' \
+            b'  <SubscribeResult>\n' \
+            b'    <SubscriptionArn>arn:aws:sns:us-east-1:123456789012:test-topic:abc</SubscriptionArn>\n' \
+            b'  </SubscribeResult>\n' \
+            b'  <ResponseMetadata>\n' \
+            b'    <RequestId>c4407779-24a4-56fa-982c-3d927f93a775</RequestId>\n' \
+            b'  </ResponseMetadata>\n' \
+            b'</SubscribeResponse>'
+        response.status_code = 200
+
+        sns_listener.do_create_topic(topic_arn)
+
+        sns = sns_listener.ProxyListenerSNS()
```

Comment:
Instead of calling the proxy listener directly here, can we make actual API calls via the `boto3` library instead:
```
sns_client = aws_stack.connect_to_service('sns')
topic_response = sns_client.create_topic(Name=...)
sns_client.subscribe(TopicArn=..., Protocol='sqs', Endpoint=..., Attributes={...})
...
```
Given that this has side effects and is more of an integration test, perhaps better to move this test to `tests/integration/test_sns.py` (see also the other examples in that file)